CaloConditions/src/CaloDAQConditionsMaker.cc

```diff
@@ -0,0 +1,57 @@
+#include "CaloConditions/inc/CaloDAQConditionsMaker.hh"
+//#include "DataProducts/inc/StrawId.hh"
+
+#include <vector>
+
+using namespace std;
+
+namespace mu2e {
+  typedef std::shared_ptr<CaloDAQConditions> ptr_t;
+
+  ptr_t CaloDAQConditionsMaker::fromFcl() {
+
+    // creat this at the beginning since it must be used,
+    // partially constructed, to complete the construction
+    auto ptr = std::make_shared<CaloDAQConditions>(_config.DIRAC2CaloMap(), _config.Calo2DIRACMap());
+
+    return ptr;
+
+  } // end fromFcl
+
+  ptr_t CaloDAQConditionsMaker::fromDb(
+      DIRACtoCalo::cptr_t tdtc,
+      CalotoDIRAC::cptr_t tctd ) {
+
+    // initially fill from fcl to get all the constants
+    auto ptr = fromFcl();
+
+    // now fill it up .. 
+    // For calorimeter local array#1: DIRAC2Calo
+    // Loops over 136*200 channels
+    //
+    int NumCaloDIRAC=136;
+    int NumChanDIRAC=20;
+    int NumDIRACTotChannel = NumCaloDIRAC*NumChanDIRAC;
```

Comment:
Following up on Ray's comment from a few days ago about hard coded numbers.   I see two options for numbers like this.

The preferred option is to put them into the calorimeter Geometry object or, if they may be time dependent into Proditions.  Be smart about getting the numbers once at the start of each event and not many times during the event.  The software engineers recommend the following: put all of these numbers into a struct and pass that struct by reference down the call chain; they prefer this over making the struct a data member.

If and only if this produces code that it is too slow, then collect all of these definitions  into one header file and declare them like
constexpr int NumChanDIRAC=20.  
A constexpr is a pure compile time concept and it produces code that is identical to putting in the hard coded number.  Put this header file in the the calorimeter Geometry inc/ area.  Also update the code that initializes the geometry to check that these numbers are consistent with the numbers read from the geometry file  ( or equivalent in Proditions if this is time dependent.)

I think it's OK to merge this PR without making this change but please do it soon afterward in a separate PR.

Reply:
We have added these parameters in the Calorimeter_csi.txt file.
We expect these numbers to be constant.
Cheers,
Stefano